Bottom: bbc82bfd2ae09f91ad708f714442493e57758ab8
Top:    c7262e9613a4f4f621f6fd80832927331e22101f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-01-23 18:50:48 +0000

Cure crosshair lagging on screen






---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index fb97baa..8b01fb7 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -998,6 +998,7 @@ ghid_port_window_motion_cb (GtkWidget * widget,
   static gint x_prev = -1, y_prev = -1;
   gboolean moved;
   GdkModifierType state;
+#if 0
   GdkGLContext* pGlContext = gtk_widget_get_gl_context (widget);
   GdkGLDrawable* pGlDrawable = gtk_widget_get_gl_drawable (widget);
 
@@ -1006,6 +1007,7 @@ ghid_port_window_motion_cb (GtkWidget * widget,
     printf ("GL THingy returned\n");
     return FALSE;
   }
+#endif
 
   state = (GdkModifierType) (ev->state);
   mk = ghid_modifier_keys_state (&state);
@@ -1013,8 +1015,11 @@ ghid_port_window_motion_cb (GtkWidget * widget,
   if ((ev->state & GDK_BUTTON3_MASK) == GDK_BUTTON3_MASK
       && mk == NONE_PRESSED)
     {
+/* Not sure this really matters, since we're using invalidate to get a redraw */
+#if 1
       if (gtk_events_pending ())
 	return FALSE;
+#endif
       dx = gport->zoom * (x_prev - ev->x);
       dy = gport->zoom * (y_prev - ev->y);
       if (x_prev > 0)
@@ -1026,10 +1031,11 @@ ghid_port_window_motion_cb (GtkWidget * widget,
   x_prev = y_prev = -1;
   moved = ghid_note_event_location (ev);
 
-  ghid_show_crosshair (TRUE);
-  if (moved && have_crosshair_attachments ())
+//  ghid_show_crosshair (TRUE);
+  if (moved) // && have_crosshair_attachments ())
     ghid_draw_area_update (gport, NULL);
 
+#if 0
   if (gdk_gl_drawable_is_double_buffered (pGlDrawable))
     gdk_gl_drawable_swap_buffers (pGlDrawable);
   else
@@ -1037,6 +1043,8 @@ ghid_port_window_motion_cb (GtkWidget * widget,
 
   /* end drawing to current GL-context */
   gdk_gl_drawable_gl_end (pGlDrawable);
+#endif
+
   return FALSE;
 }
